src/Rules/Variables/UnsetRule.php

```diff
@@ -103,6 +103,19 @@ private function canBeUnset(Node $node, Scope $scope): ?IdentifierRuleError
 					->identifier($propertyReflection->isReadOnly() ? 'unset.readOnlyProperty' : 'unset.readOnlyPropertyByPhpDoc')
 					->build();
 			}
+
+			if ($propertyReflection->isHooked()) {
+				return RuleErrorBuilder::message(
+					sprintf(
+						'Cannot unset hooked %s::$%s property.',
+						$propertyReflection->getDeclaringClass()->getDisplayName(),
+						$foundPropertyReflection->getName(),
+					),
+				)
+					->line($node->getStartLine())
+					->identifier('unset.hookedProperty')
+					->build();
```

Comment:
non-ignorable?

Reply:
Rather not, because the error depends on the type of the expr the property is fetched on. And that might be wrong, so it should be ignorable.